src/backend/access/transam/xlogrecovery.c

```diff
@@ -91,6 +91,7 @@ TimestampTz recoveryTargetTime;
 const char *recoveryTargetName;
 XLogRecPtr	recoveryTargetLSN;
 int			recovery_min_apply_delay = 0;
+bool		allowReplicaMisconfig = false;
```

Comment:
Where's the GUC for this defined? It was in guc_tables.c in an earlier version of this PR but I don't see it anymore. Is it in the neon extension perhaps? If so, let's add a comment here pointing that out.

Reply:
@MMeent  wrote that hue is against exposing such GUC in Postgres core:
https://github.com/neondatabase/postgres/pull/501#discussion_r1767022456
so I moved to extension.

Will add comment.